app/models/katello/event_queue.rb

```diff
@@ -0,0 +1,25 @@
+module Katello
+  class EventQueue < Katello::Model
+    self.table_name = 'katello_event_queue'
+
+    @repository_types = {}
+
+    def self.pop_event
+      first = Katello::EventQueue.order(:created_at => 'asc').limit(1)
+      events = Katello::EventQueue.where(:object_id => first.pluck(:object_id), :event_type => first.pluck(:event_type)).destroy_all
```

Comment:
Since this is doing a destroy when it pops them off, is this safe against listener failure and restarts?